### PR TITLE
[feat] Add triton framework for models.

### PR DESCRIPTION
**Pull request by dolaameng**
Add triton framework for models. This PR makes changes consistent with web-tier changes http://go/kaggle-pr/30587.
Will discuss with @stevemessick to see if we need to create a new release.

FIX=b/353864610
DIFFBASE=http://go/kaggle-pr/30587
Tested
- Run
```
api.model_create_new("/tmp/model_meta")
api.model_instance_version_create("limakaggle/instance1/triton/var1", "/tmp/empty")
```

### CODE REVIEW

**Comment by dolaameng** [comment]
Is there a better way to test these changes? I don't see any pytest in test/ integration_test folder though.

**Comment by neshdev** [comment]
> Add triton framework for models. This PR makes changes consistent with web-tier changes http://go/kaggle-pr/30587. Will discuss with @stevemessick to see if we need to create a new release.
> 
> FIX=b/353864610 DIFFBASE=http://go/kaggle-pr/30587 Tested
> 
> * Run
> 
> ```
> api.model_create_new("/tmp/model_meta")
> api.model_instance_version_create("limakaggle/instance1/triton/var1", "/tmp/empty")
> ```

Did those command work? The instruction's can be found [here](https://screenshot.googleplex.com/53SwdMGnUW3N6R3).
I recall that you need a metadata file for the upload.

**Comment by stevemessick** [comment]
Unfortunately, testing in this project has historically been of secondary importance. I intend to change that, but it is still on the to-do list.

**Comment by dolaameng** [comment]
> > Add triton framework for models. This PR makes changes consistent with web-tier changes http://go/kaggle-pr/30587. Will discuss with @stevemessick to see if we need to create a new release.
> > FIX=b/353864610 DIFFBASE=http://go/kaggle-pr/30587 Tested
> > 
> > * Run
> > 
> > ```
> > api.model_create_new("/tmp/model_meta")
> > api.model_instance_version_create("limakaggle/instance1/triton/var1", "/tmp/empty")
> > ```
> 
> Did those command work? The instruction's can be found [here](https://screenshot.googleplex.com/53SwdMGnUW3N6R3). I recall that you need a metadata file for the upload.

Oh yes I first run `api.model_initialize` and `api_model_instance_initialize` to create these meta files and fill them out. Sorry I didn't include the whole snippet here.

**Review by stevemessick — APPROVED** [review]
Please undo the formatting change in kaggle_api_extended.py (both copies). Formatting is tricky since it needs to be done in the container. We're using an old version of yapf.

**Comment by dolaameng** [comment]
> Please undo the formatting change in kaggle_api_extended.py (both copies). Formatting is tricky since it needs to be done in the container. We're using an old version of yapf.

I see. Thank you for the review and explanation. Done.

File: kaggle/models/model_new_instance_request.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def framework(self, framework):
         """
         if framework is None:
             raise ValueError("Invalid value for `framework`, must not be `None`")  # noqa: E501
-        allowed_values = ["tensorFlow1", "tensorFlow2", "tfLite", "tfJs", "pyTorch", "jax", "flax", "pax", "maxText", "gemmaCpp", "tensorRtLlm", "ggml", "gguf", "coral", "scikitLearn", "mxnet", "onnx", "keras", "transformers", "other"]  # noqa: E501
+        allowed_values = ["tensorFlow1", "tensorFlow2", "tfLite", "tfJs", "pyTorch", "jax", "flax", "pax", "maxText", "gemmaCpp", "tensorRtLlm", "ggml", "gguf", "coral", "scikitLearn", "mxnet", "onnx", "keras", "transformers", "triton", "other"]  # noqa: E501
```
Comment (rosbo) [inline kaggle/models/model_new_instance_request.py:162]: The files under `kaggle/` are generated (build output). I am discussing with Steve about removing them from the repo to avoid this confusion.

Some file are copied from `/src/kaggle` at build time. Others (like this one), are generated by Swagger from the KaggleSwagger.yaml configuration file.

tl;dr this change will be overridden at the next release. Updating `KaggleSwagger.yaml` was enough.

cc/ @dolaameng @stevemessick



For instance, this file is generated from the
